src/lib.rs

```diff
@@ -1024,6 +926,79 @@ impl<'a> Event<'a> {
             name  : name,
         }
     }
+
+    fn from_buffer(fd: Weak<RawFd>, buffer: &[u8], num_bytes: usize) -> Self {
```

Comment:
This currently extracts a single event per `read` call. There will need to be some buffer management in order to extract multiple events from a single `read`.